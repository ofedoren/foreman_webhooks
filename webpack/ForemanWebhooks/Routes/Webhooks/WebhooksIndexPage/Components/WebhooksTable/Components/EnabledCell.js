import React from 'react';
import PropTypes from 'prop-types';
import { CheckIcon, BanIcon } from '@patternfly/react-icons';

const EnabledCell = ({ condition }) =>
  condition ? <CheckIcon /> : <BanIcon />;

EnabledCell.propTypes = {
  condition: PropTypes.bool,
};

EnabledCell.defaultProps = {
  condition: false,
};

export default EnabledCell;
