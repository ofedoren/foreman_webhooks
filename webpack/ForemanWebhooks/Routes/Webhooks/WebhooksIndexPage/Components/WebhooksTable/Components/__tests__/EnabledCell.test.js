import { testComponentSnapshotsWithFixtures } from '@theforeman/test';
import EnabledCell from '../EnabledCell';

const fixtures = {
  'should render blank cell': {
    condition: false,
  },
  'should render marked cell': {
    condition: true,
  },
};

describe('EnabledCell', () =>
  testComponentSnapshotsWithFixtures(EnabledCell, fixtures));
