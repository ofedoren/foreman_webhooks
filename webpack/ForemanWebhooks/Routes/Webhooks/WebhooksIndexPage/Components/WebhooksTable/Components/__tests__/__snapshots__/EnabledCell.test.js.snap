// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnabledCell should render blank cell 1`] = `
<BanIcon
  color="currentColor"
  noVerticalAlign={false}
  size="sm"
/>
`;

exports[`EnabledCell should render marked cell 1`] = `
<CheckIcon
  color="currentColor"
  noVerticalAlign={false}
  size="sm"
/>
`;
